lib/widgets/screens/sign_up/sign_up_business_stage.dart

```diff
@@ -16,27 +19,80 @@ class SignupBusinessStageScreen extends StatefulWidget {
 }
 
 class _SignupBusinessStageScreenState extends State<SignupBusinessStageScreen> {
+  late final UserRegistrationModel _registrationModel;
+  int _selectedStageIndex = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    _registrationModel = Provider.of<UserRegistrationModel>(
+      context,
+      listen: false,
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
     final AppLocalizations l10n = AppLocalizations.of(context)!;
-
     return SignUpTemplate(
       progress: SignUpProgress.one,
-      title: l10n.businessStage,
+      title: l10n.signupBusinessStageTitle,
       bottomButtons: SignUpBottomButtons(
           leftButtonText: l10n.previous,
           rightButtonText: l10n.next,
           leftOnPress: () {
             context.pop();
           },
           rightOnPress: () {
+            switch (_selectedStageIndex) {
+              case 0:
+                _registrationModel.updateUserInput.companyStageTextId =
+                    CompanyStageTextId.idea.name;
+                break;
+              case 1:
+                _registrationModel.updateUserInput.companyStageTextId =
+                    CompanyStageTextId.operational.name;
+                break;
+              case 2:
+                _registrationModel.updateUserInput.companyStageTextId =
+                    CompanyStageTextId.earning.name;
+                break;
+              case 3:
+                _registrationModel.updateUserInput.companyStageTextId =
+                    CompanyStageTextId.profitable.name;
+                break;
+              default:
+                _registrationModel.updateUserInput.companyStageTextId = null;
+                break;
+            }
             context.push(Routes.signupBusinessHelpSelection.path);
           }),
       footer: SignUpIconFooter(
           icon: Icons.visibility_outlined, text: l10n.signUpShownOnProfileInfo),
       body: Column(
         children: [
-          createBusinessStageCards(context),
+          LoginRadioButtonCards(
+            title: [
+              l10n.signupBusinessStageCard1Title,
```

Comment:
Yes the reason I am not fetching these from the content provider is because of the missing descriptions. Once that's added we will definitely make the change.